File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/season/model/TvSeasonAggregateCredits.java
package dev.reuss.tmdb.domain.tv.season.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Aggregate credits for a TMDB TV season.
 *
 * @param cast aggregate cast credits
 * @param crew aggregate crew credits
 * @param id   TMDB season id
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeasonAggregateCredits(
        List<TvSeasonAggregateCastCredit> cast,
        List<TvSeasonAggregateCrewCredit> crew,
        int id
) implements TmdbModel {

    public TvSeasonAggregateCredits {
        cast = TmdbCollections.list(cast);
        crew = TmdbCollections.list(crew);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/season/model/TvSeasonAggregateCredits.kt
package dev.reuss.tmdb.domain.tv.season.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Aggregate credits for a TMDB TV season.
 *
 * @property cast aggregate cast credits
 * @property crew aggregate crew credits
 * @property id TMDB season id
 */
@JvmRecord
data class TvSeasonAggregateCredits(
    @all:JsonProperty("cast")
    val cast: List<TvSeasonAggregateCastCredit> = emptyList(),
    @all:JsonProperty("crew")
    val crew: List<TvSeasonAggregateCrewCredit> = emptyList(),
    @all:JsonProperty("id")
    val id: Int,
) : TmdbModel
